packages/@aws-cdk/aws-eks/lib/kubectl-handler/helm/__init__.py

```diff
@@ -51,6 +65,14 @@ def helm_handler(event, context):
             f.write(json.dumps(values, indent=2))
 
     if request_type == 'Create' or request_type == 'Update':
+        # Ensure chart or chart_asset_url are set
+        assert(chart != None or chart_asset_url != None)
+
+        if chart_asset_url != None and chart_asset_url.startswith('s3://') and repository == None:
+            # future work: support versions from s3 assets
+            assert(version==None)
+            chart = get_chart_asset_from_url(chart_asset_url)
```

Comment:
What if the URL doesn't start with `s3://`? What if repository is passed in? 
We shouldn't be silently falling back to the previous behavior as this suggest a misuse. 

```suggestion
        if chart_asset_url != None:

            assert(chart==None)
            assert(repository==None)
            assert(version==None)

            if not chart_asset_url.chart_asset_url.startswith('s3://'):
              raise RuntimeError(f'ChartAssetURL must point to as s3 location but is {chart_asset_url}')

            # future work: support versions from s3 assets
            chart = get_chart_asset_from_url(chart_asset_url)
```

Reply:
Updated

Reply:
Oops. Should have been `chart_asset_url.startswith('s3://')` instead of `chart_asset_url.chart_asset_url.startswith('s3://')`